docs/src/developing/clients/javascript-reference.md

```diff
@@ -0,0 +1,873 @@
+// ]
+
+
+let accountFromSecret = Keypair.fromSecretKey(account.secretKey);
+
+console.log(accountFromSecret.publicKey.toBase58());
+console.log(accountFromSecret.secretKey);
+
+// 2DVaHtcdTf7cm18Zm9VV8rKK4oSnjmTkKE6MiXe18Qsb
+// Uint8Array(64) [
+//   152,  43, 116, 211, 207,  41, 220,  33, 193, 168, 118,
+//    24, 176,  83, 206, 132,  47, 194,   2, 203, 186, 131,
+//   197, 228, 156, 170, 154,  41,  56,  76, 159, 124,  18,
+//    14, 247,  32, 210,  51, 102,  41,  43,  21,  12, 170,
+//   166, 210, 195, 188,  60, 220, 210,  96, 136, 158,   6,
+//   205, 189, 165, 112,  32, 200, 116, 164, 234
+// ]
+```
+
+Using `generate` generates a random Keypair for use as an account on Solana. Using `fromSeed`, you can generate a Keypair using a deterministic constructor. `fromSecret` creates a Keypair from a secret Uint8array. You can see that the publicKey for the `generate` Keypair and `fromSecret` Keypair are the same because the secret from the `generate` Keypair is used in `fromSecret`.
+
+**Warning**: Do not use `fromSeed` unless you are creating a seed with high entropy. Do not share your seed. Treat the seed like you would a private key.
+
+### Loader
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/Loader.html)
+
+Loader is used in [BpfLoader](javascript-api.md#BpfLoader) to load rust compiled programs on chain.
+
+Please refer to [BpfLoader](javascript-api.md#Loader) for usage.
+
+### Lockup
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/Lockup.html)
+
+Lockup is used in conjunction with the [StakeProgram](javascript-api.md#StakeProgram) to create an account. The Lockup is used to determine how long the stake will be locked, or unable to be retrieved. If the Lockup is set to 0 for both epoch and the Unix timestamp, the lockup will be disabled for the stake account.
+
+#### Example Usage
+
+```javascript
+const {Authorized, Keypair, Lockup, StakeProgram} = require("@solana/web3.js");
+
+let account = Keypair.generate();
+let stakeAccount = Keypair.generate();
+let authorized = new Authorized(account.publicKey, account.publicKey);
+let lockup = new Lockup(0, 0, account.publicKey);
+
+let createStakeAccountInstruction = StakeProgram.createAccount({
+    fromPubkey: account.publicKey,
+    authorized: authorized,
+    lamports: 1000,
+    lockup: lockup,
+    stakePubkey: stakeAccount.publicKey
+});
+```
+The above code creates a `createStakeAccountInstruction` to be used when creating an account with the `StakeProgram`. The Lockup is set to 0 for both the epoch and Unix timestamp, disabling lockup for the account.
+
+See [StakeProgram](javascript-api.md#StakeProgram) for more.
+
+### Message
+
+[Source Documentation](https://solana-labs.github.io/solana-web3.js/classes/Message.html)
+
+Message is used as another way to construct transactions. You can construct a message using the accounts, header, instructions, and recentBlockhash that are a part of a transaction.
```

Comment:
Can you also say that a transaction is the combination of a message and a list of signatures? It's not too clear that `Message` is basically the same thing as a transaction (minus the signatures)